### PR TITLE
feat(store): resize PackedCounter to use 5 bytes per length

### PR DESCRIPTION
**Pull request by dk1a**
fixes https://github.com/latticexyz/mud/issues/798

### CODE REVIEW

File: packages/network/src/v2/schemas/decodeData.ts [hunk 1]
```diff
@@ -29,13 +29,20 @@ export const decodeData = (schema: TableSchema, hexData: string): Record<number,
...
+    const packedCounterCounterType = SchemaType.UINT40;
+    const dynamicDataLength = decodeStaticField(packedCounterAccumulatorType, dynamicDataLayout, 0) as number;
 
     schema.dynamicFields.forEach((fieldType, i) => {
-      const dataLength = dynamicDataLayout.getUint16(4 + i * 2);
+      const dataLength = decodeStaticField(
+        packedCounterCounterType,
+        dynamicDataLayout,
+        getStaticByteLength(packedCounterAccumulatorType) + i * getStaticByteLength(packedCounterCounterType)
+      ) as number;
```
Comment (holic) [inline packages/network/src/v2/schemas/decodeData.ts:45]: haven't actually tested this yet! please test before merging!

I assume that before these client changes, array values wouldn't decode properly (they'd be sliced in the wrong places). These changes should address that.

File: packages/cli/src/commands/deploy.ts [hunk 2]
```diff
@@ -51,64 +26,8 @@ export const yDeployOptions = {
...
-      console.log(
-        chalk.bgGreen(
-          chalk.whiteBright(`\n Deployment result (written to ${mudConfig.worldsFile} and ${outputDir}): \n`)
-        )
-      );
-    }
-
-    console.log(deploymentInfo);
-    return deploymentInfo;
+    deployHandler(args);
```
Comment (alvrs) [inline packages/cli/src/commands/deploy.ts:30]: this seems to be an infinite recursion now? I guess we want to import `deployHandler` from `deployHandler`?

File: packages/config/src/node/loadConfig.ts [hunk 3]
```diff
@@ -15,7 +15,7 @@ export async function loadConfig(configPath?: string): Promise<unknown> {
     configPath = await resolveConfigPath(TEMP_CONFIG);
     return (await import(configPath)).default;
   } finally {
-    rmSync(TEMP_CONFIG);
+    rmSync(TEMP_CONFIG, { force: true });
```
Comment (alvrs) [inline packages/config/src/node/loadConfig.ts:18]: why is `force: true` needed here?

File: packages/cli/src/commands/deploy.ts (hunk 2 above)
Comment (dk1a) [reply packages/cli/src/commands/deploy.ts]: looks like I broke it while merging, tried to fix, can test some time later today/tomorrow

File: packages/config/src/node/loadConfig.ts (hunk 3 above)
Comment (dk1a) [reply packages/config/src/node/loadConfig.ts:18]: if something within `try` throws, the TEMP_CONFIG may not exist, which will cause rmSync to throw, which will swallow the original error. force makes rmSync do nothing if the file doesnt exist

File: packages/cli/src/commands/deploy.ts [hunk 4]
```diff
@@ -125,7 +35,12 @@ const commandModule: CommandModule<DeployOptions, DeployOptions> = {
   },
 
   async handler(args) {
-    await deployHandler(args);
+    try {
+      deployHandler(args);
```
Comment (holic) [inline packages/cli/src/commands/deploy.ts:39]: should this `await`?

Comment (alvrs) [reply packages/cli/src/commands/deploy.ts]: it should, fixed it

**Review by alvrs — APPROVED** [review]

**Review by alvrs — DISMISSED** [review]

**Review by alvrs — APPROVED** [review]

**Pull request merged**